ethereum/contracts/bridgehub/bridgehub-deps/BridgehubAdmin.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "../bridgehub-interfaces/IBridgehubAdmin.sol";
+import "../../common/libraries/Diamond.sol";
+import "../../common/libraries/L2ContractHelper.sol";
+import "./BridgehubBase.sol";
+
+/// @title Admin Contract controls access rights for contract management.
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+contract BridgehubAdminFacet is BridgehubBase, IBridgehubAdmin {
+    string public constant override getName = "BridgehubAdminFacet";
+
+    /// @notice Starts the transfer of governor rights. Only the current governor can propose a new pending one.
+    /// @notice New governor can accept governor rights by calling `acceptGovernor` function.
+    /// @param _newPendingGovernor Address of the new governor
+    function setPendingGovernor(address _newPendingGovernor) external onlyGovernor {
+        // Save previous value into the stack to put it into the event later
+        address oldPendingGovernor = bridgehubStorage.pendingGovernor;
+        // Change pending governor
+        bridgehubStorage.pendingGovernor = _newPendingGovernor;
+        emit NewPendingGovernor(oldPendingGovernor, _newPendingGovernor);
+    }
+
+    /// @notice Accepts transfer of governor rights. Only pending governor can accept the role.
+    function acceptGovernor() external {
+        address pendingGovernor = bridgehubStorage.pendingGovernor;
+        require(msg.sender == pendingGovernor, "n4"); // Only proposed by current governor address can claim the governor rights
+
+        address previousGovernor = bridgehubStorage.governor;
+        bridgehubStorage.governor = pendingGovernor;
+        delete bridgehubStorage.pendingGovernor;
+
+        emit NewPendingGovernor(pendingGovernor, address(0));
+        emit NewGovernor(previousGovernor, pendingGovernor);
+    }
+
+    /// @notice Starts the transfer of admin rights. Only the current governor or admin can propose a new pending one.
+    /// @notice New admin can accept admin rights by calling `acceptAdmin` function.
+    /// @param _newPendingAdmin Address of the new admin
+    function setPendingAdmin(address _newPendingAdmin) external onlyGovernorOrAdmin {
+        // Save previous value into the stack to put it into the event later
+        address oldPendingAdmin = bridgehubStorage.pendingAdmin;
+        // Change pending admin
+        bridgehubStorage.pendingAdmin = _newPendingAdmin;
+        emit NewPendingGovernor(oldPendingAdmin, _newPendingAdmin);
```

Comment:
the event is incorrect. You should merge `dev` branch into yours one

Reply:
Bridgehub is now TUP, not relevant. (I have merged dev, but it is a separate contract so it did not merge this)